### ISSUE TITLE
tsconfig paths with a '.' in the path target break next.config.ts transpilation

### ISSUE THREAD

**Issue by jmikrut** [vercel/next.js#70711]
### Link to the code that reproduces this issue

https://github.com/jmikrut/next-config-ts-paths

### To Reproduce

1. `npm i`
2. `npm run dev`, see that it works
3. go to `next.config.ts` and comment out `import { hello } from '@hello-works'`
4. then comment _in_ `import { hello } from '@hello-broken'`
5. `npm run dev`
6. boom

Looks like paths with a `.` in the name break the transpilation `next.config.ts`. 

Of note, paths that have `.` in the name work fine outside of `next.config.ts`.

### Current vs. Expected behavior

Currently, paths in `tsconfig.json` with `.` in the name break transpilation of the `next.config.ts` file.

I expect them to work as they do anywhere else in the app.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.0.0: Mon Aug 12 20:51:54 PDT 2024; root:xnu-11215.1.10~2/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 20.11.1
  npm: 10.2.4
  Yarn: 1.16.0
  pnpm: 9.8.0
Relevant Packages:
  next: 15.0.0-canary.176 // Latest available version is detected (15.0.0-canary.176).
  eslint-config-next: N/A
  react: 19.0.0-rc-2d16326d-20240930
  react-dom: 19.0.0-rc-2d16326d-20240930
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

TypeScript

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local)

### Additional context

The error is:

```
Error: Cannot find module './app/this'
```

The imported module name should be `./app/this.doesnt.ts`, but it looks like the module name is getting chopped off at the `.`.

**Comment by samcx** [comment]
@jmikrut Thank you for submitting an issue!

I was indeed able to replicate this. I will share this with the team and take a look!

**Comment by jmikrut** [comment]
@samcx of course. keep up the good work and thank you for looking into it!

**Comment by devjiwonchoi** [comment]
Hi, thank you for reporting the issue! We use SWC to resolve `paths`, and had a relevant fix at https://github.com/swc-project/swc/pull/9595 that need to be shipped and applied to next repo as well. Will work on it ASAP when it gets shipped. Thank you!

**Comment by devjiwonchoi** [comment]
Fixed by #70687
